upstream.wrap

```diff
@@ -0,0 +1,7 @@
+[wrap-file]
+directory = nlohmann_json-3.9.1
+lead_directory_missing = true
+
+source_url = https://github.com/nlohmann/json/releases/download/v3.9.1/include.zip
+source_filename = nlohmann_json-3.9.1.zip
+source_hash = 6bea5877b1541d353bd77bdfbdb2696333ae5ed8f9e8cc22df657192218cad91
```

Comment:
You should have a `[provide]` section as described here:

https://mesonbuild.com/Wrap-dependency-system-manual.html#provide-section

Reply:
I see, I have added the `[provide]` section as requested.

Reply:
`dependency_names` assumes the meson.build uses override_dependency, which nlohmann_json doesn't. Instead you'll need to define the dependency name as equal to the nlohmann_json_dep you're accessing.

Reply:
So it should be:

```
[provide]
dependency_names = nlohmann_json_dep
```

Is it correct? Sorry but this part is pretty new and a bit confusing.

Reply:
> This mechanism assumes the subproject calls `meson.override_dependency('foo-1.0', foo_dep)` so Meson knows which dependency object should be used as fallback. Since that method was introduced in version 0.54.0, as a transitional aid for projects that do not yet make use of it the variable name can be provided in the wrap file with entries in the format `foo-1.0 = foo_dep`.

Reply:
I see. Thanks for the explanation!